base/precompilation.jl

```diff
@@ -817,7 +825,16 @@ function _precompilepkgs(pkgs::Vector{String},
                             t = @elapsed ret = precompile_pkgs_maybe_cachefile_lock(io, print_lock, fancyprint, pkg_config, pkgspidlocked, hascolor) do
                                 Base.with_logger(Base.NullLogger()) do
                                     # The false here means we ignore loaded modules, so precompile for a fresh session
-                                    Base.compilecache(pkg, sourcepath, std_pipe, std_pipe, false; flags, cacheflags, isext = haskey(exts, pkg))
+                                    keep_loaded_modules = false
+                                    if haskey(exts, pkg)
+                                        # any extension in our direct dependencies is one we have a right to load
+                                        loadable_exts = filter((dep)->haskey(exts, dep), depsmap[pkg])
+                                        Base.compilecache(pkg, sourcepath, std_pipe, std_pipe, keep_loaded_modules;
+                                                          flags, cacheflags, loadable_exts)
+                                    else
+                                        Base.compilecache(pkg, sourcepath, std_pipe, std_pipe, keep_loaded_modules;
```

Comment:
Seems easier to have one call to compilecache where loadable_exts is empty in the other case.

Reply:
FWIW `loadable_exts === nothing` is the opposite of an empty list - it actually gives you the right to load any extension (because you are a package)